File: internal/astgrep/testdata/fixtures/elixir/valid.ex
# Elixir valid fixture — no rule violations.
# This file should produce 0 findings when scanned with the elixir rule set.

defmodule Example do
  def good_code do
    result = :some_value
    {:ok, value} = Map.fetch(%{key: "val"}, :key)
    value
  end

  def implemented do
    {:ok, "result"}
  end
end

File: internal/astgrep/testdata/fixtures/csharp/valid.cs
// C# valid fixture — no rule violations.
// This file should produce 0 findings when scanned with the csharp rule set.

using System;

public class Example
{
    public void GoodCode()
    {
        string x = "hello";
        var s = Convert.ToString(x);
        Console.WriteLine(s);
    }

    public string Implemented()
    {
        return "actual result";
    }
}
